### ISSUE TITLE
When linting a .cljc file, include platform(s) in issues reported

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1196]
**Is your feature request related to a problem? Please describe.**
When I use clj-kondo to lint a `.cljc` file I sometimes get confused.
Clj-kondo does not currently report for which platform(s) it discovered each linting issue.

If the lint issue is for all platforms my mind is clear, I know what to fix.
If the lint issue is for just one platform (usually it ends up being cljs), it can take me sometimes a while to remember I am in an `.cljc` file and that I am dealing with just a cljs problem vs a cljc problem.

**Describe the solution you'd like**
Let's take a contrived `example.cljc` file:

```Clojure
(ns example)

;; oh happy times
(+ 1 2 3)

;; I've made a cross platform mistake here blapperflap is not available anywhere
(blapperflap)

;; I've made a cljs mistake here, format is not available in cljs
(format "bing bang %s" "bong")

#?(:cljs
   (defn yabbadabba []))

;; I've made a clj mistake here, yabbadabba is only available in cljs
(yabbadabba)
```

If I run clj-kondo v2021.03.03 on `example.cljc` I get the following report:
```Shell
$ clj-kondo --lint example.cljc
example.cljc:7:2: error: Unresolved symbol: blapperflap
example.cljc:10:2: error: Unresolved symbol: format
example.cljc:16:2: error: Unresolved symbol: yabbadabba
linting took 16ms, errors: 3, warnings: 0
```

It is doing its job of reporting issues, but I'd also love to see platform information, maybe something like:
```Shell
example.cljc:7:2: error [clj,cljs]: Unresolved symbol: blapperflap
example.cljc:10:2: error [cljs]: Unresolved symbol: format
example.cljc:16:2: error [clj]: Unresolved symbol: yabbadabba
```

When I first thought about this I liked the idea of not reporting anything when all platforms were involved.
For example:
```Shell
example.cljc:7:2: error: Unresolved symbol: blapperflap
example.cljc:10:2: error [cljs]: Unresolved symbol: format
example.cljc:16:2: error [clj]: Unresolved symbol: yabbadabba
```
But maybe always reporting all platforms offers us consistency and removes any mystery.

This thinking means the platform would still reported for `.clj` file when it can only be `clj` and a `.cljs` file when it can only be `.cljs`.

**Describe alternatives you've considered**
I'll explore more on request. Lemme know.

**Additional context**
My platform head-scratching does occur at the terminal, but mostly it happens when I'm in my editor.  
Each editor integration would have to decide how to best present the platform info.

**Comment by borkdude** [comment]
There is one problem with the format `example.cljc:16:2: error [clj]: Unresolved symbol: yabbadabba`: this isn't the standard format many linters will expect. A better one will be:

```
example.cljc:16:2: error: Unresolved symbol: yabbadabba [clj]
``` 

We could even add the linting keyword there too. 

```
example.cljc:16:2: error: Unresolved symbol: yabbadabba [:clj, :unresolved-symbol]
```

**Comment by lread** [comment]
To recap a chat on Slack:
- I think the suffix idea makes sense.
- I asked about what would happen if 2 platforms were involved in an issue.  This would result in 2 issues.

So from what I understand, looking at my original `example.clj`, at the terminal we could get output like:
```Shell
example.cljc:7:2: error: Unresolved symbol: blapperflap [clj]
example.cljc:7:2: error: Unresolved symbol: blapperflap [cljs]
example.cljc:10:2: error: Unresolved symbol: format [cljs]
example.cljc:16:2: error: Unresolved symbol: yabbadabba [clj]
```

With an `:ouput` `:suffix` of `[:language]`.

**Comment by borkdude** [comment]
Correct

**Comment by borkdude** [comment]
Reopen
